stdlib/public/Reflection/CMakeLists.txt

```diff
@@ -53,4 +53,8 @@ if(SWIFT_INCLUDE_TOOLS)
     set(CMAKE_C_COMPILER ${CURRENT_CMAKE_C_COMPILER})
     set(CMAKE_CXX_COMPILER ${CURRENT_CMAKE_CXX_COMPILER})
   endif()
+
+  if(TARGET clang)
+    add_dependencies(swiftReflection clang)
```

Comment:
Why this dependency on `clang`?  It shouldn't be needed, since CMake will depend on the compiler implicitly.

Reply:
Apparently, this target uses the just-built Clang, while everything else is built with my host compiler.

Reply:
We have two copies of swiftReflection - the first one is used for the toolchain and is built with the host compiler, the second is used for the runtime and can be built with either the just built compiler *or* the host compiler (`SWIFT_USE_HOST_COMPILER` controls that).  I think that the proper place to add this dependency is in `AddSwift.cmake` (normally, CMake would take care of this dependency itself).

Reply:
`SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER`? Hm that sounds interesting. I should probably set this in the cache.